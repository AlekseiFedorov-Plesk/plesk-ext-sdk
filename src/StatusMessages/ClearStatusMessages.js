// Copyright 1999-2018. Plesk International GmbH. All rights reserved.

import { Component, PropTypes } from '@plesk/ui-library';
import { withStatusMessages } from './index';

export class ClearStatusMessages extends Component {
    static propTypes = {
        statusMessages: PropTypes.shape({
            clear: PropTypes.func.isRequired,
        }).isRequired,
        when: PropTypes.func.isRequired,
    };

    componentDidUpdate(prevProps) {
        const { when, statusMessages } = this.props;

        if (when({ props: this.props, prevProps })) {
            statusMessages.clear();
        }
    }

    render() {
        return null;
    }
}

export default withStatusMessages(ClearStatusMessages);
