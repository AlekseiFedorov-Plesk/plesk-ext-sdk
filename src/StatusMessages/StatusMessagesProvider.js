// Copyright 1999-2018. Plesk International GmbH. All rights reserved.

import { Component, PropTypes } from '@plesk/ui-library';
import StatusMessagesProxy from './StatusMessagesProxy';

export default class ConfigProvider extends Component {
    static propTypes = {
        children: PropTypes.node,
        value: PropTypes.object,
    };

    static childContextTypes = {
        statusMessages: PropTypes.object,
    };

    static defaultProps = {
        children: null,
        value: new StatusMessagesProxy(),
    };

    getChildContext() {
        return {
            statusMessages: this.props.value,
        };
    }

    render() {
        return this.props.children;
    }
}
