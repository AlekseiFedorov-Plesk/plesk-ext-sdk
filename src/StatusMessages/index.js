// Copyright 1999-2018. Plesk International GmbH. All rights reserved.

export { default } from './StatusMessages';
export { default as StatusMessagesContext } from './StatusMessagesContext';
export { default as withStatusMessages } from './withStatusMessages';
