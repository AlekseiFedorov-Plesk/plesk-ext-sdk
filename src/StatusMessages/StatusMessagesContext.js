// Copyright 1999-2018. Plesk International GmbH. All rights reserved.

import { createContext } from '@plesk/ui-library';
import StatusMessages from './StatusMessages';

const StatusMessagesContext = createContext(
    new StatusMessages()
);

export default StatusMessagesContext;
