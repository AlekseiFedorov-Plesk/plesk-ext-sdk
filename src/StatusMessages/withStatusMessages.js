// Copyright 1999-2018. Plesk International GmbH. All rights reserved.

import { createElement } from '@plesk/ui-library';
import StatusMessagesContext from './StatusMessagesContext';

const withStatusMessages = Component => {
    const C = props => (
        <StatusMessagesContext.Consumer>
            {statusMessages => (
                <Component statusMessages={statusMessages} {...props} />
            )}
        </StatusMessagesContext.Consumer>
    );

    C.displayName = `withStatusMessages(${Component.displayName || Component.name})`;
    C.WrappedComponent = Component;

    return C;
};

export default withStatusMessages;
