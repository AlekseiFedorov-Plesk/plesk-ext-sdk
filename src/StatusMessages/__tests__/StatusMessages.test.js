// Copyright 1999-2018. Plesk International GmbH. All rights reserved.

import StatusMessages from '../StatusMessages';

describe('StatusMessages', () => {
    const addParams = [{ intent: 'success', message: 'Message' }];
    const updateParams = ['toast-1', { message: 'New message' }];
    const removeParams = ['toast-1'];

    it('call toaster methods', () => {
        const toaster = {
            add: jest.fn(),
            update: jest.fn(),
            remove: jest.fn(),
            clear: jest.fn(),
        };
        const statusMessages = new StatusMessages();

        statusMessages.setToaster(toaster);

        statusMessages.add(...addParams);
        expect(toaster.add).toHaveBeenCalledWith(...addParams);

        statusMessages.update(...updateParams);
        expect(toaster.update).toHaveBeenCalledWith(...updateParams);

        statusMessages.remove(...removeParams);
        expect(toaster.remove).toHaveBeenCalledWith(...removeParams);

        statusMessages.clear();
        expect(toaster.clear).toHaveBeenCalled();
    });

    it('prevent calls if the toaster is not setted', () => {
        const statusMessages = new StatusMessages();

        statusMessages.add(...addParams);
        statusMessages.update(...updateParams);
        statusMessages.remove(...removeParams);
        statusMessages.clear();
    });
});
