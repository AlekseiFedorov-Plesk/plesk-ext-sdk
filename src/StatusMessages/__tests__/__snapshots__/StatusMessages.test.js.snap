// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusMessages renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <StatusMessages
    location={
      Object {
        "pathname": "/",
      }
    }
    statusMessages={
      StatusMessagesProxy {
        "setToaster": [Function],
        "toaster": null,
      }
    }
  >
    <div />
  </StatusMessages>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <div />,
        <Toaster
          baseClassName="pul-toaster"
          className={null}
          position="top"
          toasts={null}
        />,
      ],
      "value": StatusMessagesProxy {
        "setToaster": [Function],
        "toaster": null,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "baseClassName": "pul-toaster",
          "className": null,
          "position": "top",
          "toasts": null,
        },
        "ref": [Function],
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.provider),
      "_context": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": [Circular],
        "Provider": [Circular],
        "_calculateChangedBits": null,
        "_changedBits": 0,
        "_changedBits2": 0,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": StatusMessagesProxy {
          "setToaster": [Function],
          "toaster": null,
        },
        "_currentValue2": StatusMessagesProxy {
          "setToaster": [Function],
          "toaster": null,
        },
        "_defaultValue": StatusMessagesProxy {
          "setToaster": [Function],
          "toaster": null,
        },
      },
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <div />,
          <Toaster
            baseClassName="pul-toaster"
            className={null}
            position="top"
            toasts={null}
          />,
        ],
        "value": StatusMessagesProxy {
          "setToaster": [Function],
          "toaster": null,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "baseClassName": "pul-toaster",
            "className": null,
            "position": "top",
            "toasts": null,
          },
          "ref": [Function],
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.provider),
        "_context": Object {
          "$$typeof": Symbol(react.context),
          "Consumer": [Circular],
          "Provider": [Circular],
          "_calculateChangedBits": null,
          "_changedBits": 0,
          "_changedBits2": 0,
          "_currentRenderer": null,
          "_currentRenderer2": null,
          "_currentValue": StatusMessagesProxy {
            "setToaster": [Function],
            "toaster": null,
          },
          "_currentValue2": StatusMessagesProxy {
            "setToaster": [Function],
            "toaster": null,
          },
          "_defaultValue": StatusMessagesProxy {
            "setToaster": [Function],
            "toaster": null,
          },
        },
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
